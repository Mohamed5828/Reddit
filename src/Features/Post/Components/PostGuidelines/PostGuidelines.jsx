// Import react bootstrap components
import Table from "react-bootstrap/Table";

// Import styled components
import {
  Container,
  PostLogo,
  TableData,
  TableHeader,
} from "./PostGuidelines.styled";

// Import images
import Guideline from "Features/Post/Assets/Images/Post-guidelines.png";

/**
 * Post guidelines component (The post guidelines in the sidebar in the create post page)
 *
 * @returns {React.Component} PostGuidelines component (The post guidelines in the sidebar in the create post page)
 */
const PostGuidelines = () => {
  return (
    <Container>
      <Table>
        <thead>
          <tr>
            <TableHeader>
              <PostLogo src={Guideline} alt="post guidelines" />
              Posting to Reddit
            </TableHeader>
          </tr>
        </thead>
        <tbody>
          <tr>
            <TableData>1. Remember the human</TableData>
          </tr>
          <tr>
            <TableData>2. Behave like you would in real life</TableData>
          </tr>
          <tr>
            <TableData>3. Look for the original source of content</TableData>
          </tr>
          <tr>
            <TableData>4. Search for duplicates before posting</TableData>
          </tr>
          <tr>
            <TableData>5. Read the communityâ€™s rules</TableData>
          </tr>
        </tbody>
      </Table>
    </Container>
  );
};

export default PostGuidelines;
