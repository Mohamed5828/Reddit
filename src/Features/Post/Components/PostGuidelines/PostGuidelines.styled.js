// Import styled
import styled from "styled-components";

// Main container for the post guidelines component
export const Container = styled.div`
  margin-top: 15px;
  background-color: ${({ theme }) => theme.background.primary};
  padding: 12px;
  margin-top: 12px;
  border-radius: 4px;
  width: 312px;

  tr {
    color: ${({ theme }) => theme.color.primary};
  }
`;

// Data item for the post guidelines table
export const TableData = styled.td`
  font-size: 14px;
  font-weight: 500;
`;

// Header item for the post guidelines table
export const TableHeader = styled.th`
  font-size: 18px;
  font-weight: 500;
  display: inline-block;
  width: 100%;
`;

// Image (logo) for the post guidelines component
export const PostLogo = styled.img`
  height: 40px;
  margin-right: 8px;
  padding-bottom: 5px;
  width: 40px;
`;
