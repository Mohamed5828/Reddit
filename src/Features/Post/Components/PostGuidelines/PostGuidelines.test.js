import { shallow } from "enzyme";
import React from "react";

// Import components
import PostGuidelines from "./PostGuidelines";

describe("Post guidelines footer", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<PostGuidelines />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
