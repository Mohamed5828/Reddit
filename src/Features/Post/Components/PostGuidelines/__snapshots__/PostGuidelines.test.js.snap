// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post guidelines footer should render without crashing 1`] = `
<styled.div>
  <ForwardRef>
    <thead>
      <tr>
        <styled.th>
          <styled.img
            alt="post guidelines"
            src="Post-guidelines.png"
          />
          Posting to Reddit
        </styled.th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <styled.td>
          1. Remember the human
        </styled.td>
      </tr>
      <tr>
        <styled.td>
          2. Behave like you would in real life
        </styled.td>
      </tr>
      <tr>
        <styled.td>
          3. Look for the original source of content
        </styled.td>
      </tr>
      <tr>
        <styled.td>
          4. Search for duplicates before posting
        </styled.td>
      </tr>
      <tr>
        <styled.td>
          5. Read the communityâ€™s rules
        </styled.td>
      </tr>
    </tbody>
  </ForwardRef>
</styled.div>
`;
